node/service/src/lib.rs

```diff
@@ -823,20 +838,95 @@ where
 					let author_id = author_id.clone();
+					let channel = downward_xcm_receiver.clone();
 
 					async move {
 						let time = sp_timestamp::InherentDataProvider::from_system_time();
 
-						let mocked_parachain = MockValidationDataInherentDataProvider {
+						//TODO maybe we don't need to collect into a Vec here. We could just
+						// leave it as an iterator.
+						let downward_messages: Vec<InboundDownwardMessage> = channel
+							.drain()
+							.map(|(message, response_channel)| {
+								// Acknowledge receipt of the message and send it to be
+								// included in the block
+								//TODO We have the parent block hash and the current block number. We
+								// won't know this block's hash until after it has been created
+								response_channel.send(true).expect("Couldn't send acknowledgement of downward xcm message to RPC layer.");
+								println!("{:?}", message);
+								message
+							})
+							.collect();
+
+						// Construct a simple downward transfer message to serialize and insert
+						// I believe this should be a `VersionedXcm<T::Call>`.
+						// Since it is generic over a runtime, I may have trouble constructing it here
+						// I could make a helper function in a runtime...
+						// Oh, unless that type info is only necessary for certain kinds of calls, and I
+						// can just put () for now...
+						let downward_transfer_message = xcm::VersionedXcm::<()>::V1(
+							xcm::v1::Xcm::<()>::ReserveAssetDeposited {
+								assets: xcm::v1::MultiAssets::from(vec![xcm::v1::MultiAsset{
+									id: xcm::v1::AssetId::Concrete(
+										xcm::v1::MultiLocation {
+											parents: 0,
+											interior: xcm::v1::Junctions::Here,
+										}
+									),
+									fun: xcm::v1::Fungibility::Fungible(10000000000000),
+								}]),
+								effects: vec![
+									// @girazoki I can't figure out how to make a literal `Order` because
+									// the variants are not public
+									xcm::v1::Order::<()>::DepositAsset{
+										assets: xcm::v1:MultiAssetFilter::Wild(xcm::v1::WildMultiAsset::All),
+										max_assets: 100, //Doesn't really matter. We only have 1
+										beneficiary: xcm::v1::MultiLocation {
+											parents: 0,
+											interior: xcm::v1::Junctions::X1(
+												xcm::v1::Junction::AccountKey20{
+													network: xcm::v1::NetworkId::Any,
+													// Alith
+													key: hex_literal::hex!("f24FF3a9CF04c71Dbc94D0b566f7A27B94566cac"),
+												}
+											),
+										},
+									}
+								],
+							}
+						);
+
+						// Here we inject our single hard-coded downward transfer message
+						downward_messages.push(
+							InboundDownwardMessage{
+								//TODO is sent_at supposed to be a realy block number or a para block number?
+								// I think it should be relay chain block number...
+								sent_at: current_para_block,//TODO
+								msg: downward_transfer_message.encode(),
+							}
+						);
```

Comment:
@girazoki Here I try to construct a simple test message. I think I got pretty close, but I can't figure out how to construct an `Order`.

My goal was that after we had this message constructed, we could trace its execution through the runtime, and determine what needs to be done to make these blocks valid.